ui/components/src/TimeSeriesTooltip/focused-series.ts

```diff
@@ -94,28 +103,65 @@ export function getNearbySeries(
                   y: yValue,
                   formattedY: formattedY,
                   markerColor: markerColor.toString(),
+                  isClosestToCursor,
                 });
+                focusedSeriesIndexes.push(seriesIdx);
               }
             }
           }
         }
       }
     }
   }
+  if (chart?.dispatchAction !== undefined) {
+    // Clears emphasis state of all lines that are not focused or emphasized.
+    // Focused is term for all nearby series showing in the tooltip.
+    // Emphasized is a subset of just the focused series that are closest to cursor.
+    chart.dispatchAction({
+      type: 'downplay',
+      seriesIndex: nonEmphasizedSeriesIndexes,
+    });
+
+    // https://echarts.apache.org/en/api.html#action.highlight
+    if (emphasizedSeriesIndexes.length > 0) {
+      // Fadeout opacity of all series not closest to cursor.
+      chart.dispatchAction({
+        type: 'highlight',
+        seriesIndex: emphasizedSeriesIndexes,
+        notBlur: false,
+      });
+    } else {
+      // When no emphasized series with bold text, notBlur allows opacity fadeout to not trigger.
+      chart.dispatchAction({
+        type: 'highlight',
+        seriesIndex: focusedSeriesIndexes,
+        notBlur: true,
+      });
+    }
+  }
+
   return currentFocusedData;
 }
 
 /**
  * Uses mouse position to determine whether user is hovering over a chart canvas
  * If yes, convert from pixel values to logical cartesian coordinates and return all focused series
  */
-export function getFocusedSeriesData(
-  mousePos: CursorData['coords'],
-  chartData: EChartsDataFormat,
-  pinnedPos: CursorData['coords'],
-  chart?: EChartsInstance,
-  unit?: UnitOptions
-) {
+export function getFocusedSeriesData({
+  mousePos,
+  chartData,
+  pinnedPos,
+  chart,
+  unit,
+  showAllSeries = false,
+}: {
+  mousePos: CursorData['coords'];
+  chartData: EChartsDataFormat;
+  pinnedPos: CursorData['coords'];
+  chart?: EChartsInstance;
+  unit?: UnitOptions;
+  showAllSeries?: boolean;
+}) {
```

Comment:
Such a minor thing, could put `chart` and `chartData` near each other since they are related ECharts things